app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.oppia.app.player.state
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.clearText
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withHint
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.home.HomeActivity
+import org.oppia.domain.exploration.TEST_EXPLORATION_ID_5
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+class StateFragmentPresenterTest {
+  @Inject
+  lateinit var stateFragmentPresenter: StateFragmentPresenter
+
+  @Test
+  fun testNumberInputView_displaySoftInputAccordingToInputType_userEnterNumber() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      val placeholder = "Write digit here."
+      onView(withHint(placeholder)).check(matches(isDisplayed()))
+      onView(withHint(placeholder)).perform(click())
+      onView(withHint(placeholder)).perform(clearText(), typeText("8"))
+    }
+  }
```

Comment:
It would good if we can add few more tests.

1. what happens to edittext/placeholder on configuration change.
2. On button click does the dummy_text_view displays the user entered string.
3. Once the dummy_text_view is set with the user entered string, what happens on configuration change now.

Reply:
resolved